src/tools/map_extractor/System.cpp

```diff
@@ -1025,6 +1026,48 @@ void ExtractDBCFiles(int locale, bool basicLocale)
     printf("Extracted %u DBC files\n\n", count);
 }
 
+void ExtractCameraFiles(int locale, bool basicLocale)
+{
+    printf("Extracting camera files...\n");
+
+    std::set<std::string> camerafiles;
+
+    // get camera file list
+    for (ArchiveSet::iterator i = gOpenArchives.begin(); i != gOpenArchives.end(); ++i)
+    {
+        std::vector<std::string> files;
+        (*i)->GetFileListTo(files);
+        for (std::vector<std::string>::iterator iter = files.begin(); iter != files.end(); ++iter)
+            if (iter->find("Cameras") == 0 && iter->rfind(".m2") == iter->length() - strlen(".m2"))
+                camerafiles.insert(*iter);
+    }
+
+    std::string path = output_path;
+    path += "/Cameras/";
+    CreateDir(path);
+    if (!basicLocale)
+    {
+        path += langs[locale];
+        path += "/";
+        CreateDir(path);
+    }
+
+    // extract M2s
+    uint32 count = 0;
+    for (std::set<std::string>::iterator iter = camerafiles.begin(); iter != camerafiles.end(); ++iter)
+    {
+        std::string filename = path;
+        filename += (iter->c_str() + strlen("Cameras\\"));
+
+        if (boost::filesystem::exists(filename))
+            continue;
+
+        if (ExtractFile(iter->c_str(), filename))
+            ++count;
+    }
+    printf("Extracted %u camera files\n\n", count);
```

Comment:
`\n\n`

I know that it's just visual but I think that you could use it just one time.

`\n`